internal/annotators/http/pki.go

```diff
@@ -0,0 +1,101 @@
+/*******************************************************************************
+ * Copyright 2022 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package http
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/project-alvarium/alvarium-sdk-go/internal/annotators"
+	"github.com/project-alvarium/alvarium-sdk-go/internal/signprovider"
+	"github.com/project-alvarium/alvarium-sdk-go/internal/signprovider/ed25519"
+	"github.com/project-alvarium/alvarium-sdk-go/pkg/config"
+	"github.com/project-alvarium/alvarium-sdk-go/pkg/contracts"
+	"github.com/project-alvarium/alvarium-sdk-go/pkg/interfaces"
+)
+
+// HttpPkiAnnotator is used to validate whether the signature on a given piece of data is valid, both sent in the HTTP message
+type HttpPkiAnnotator struct {
+	hash contracts.HashType
+	kind contracts.AnnotationType
+	sign config.SignatureInfo
+}
+
+func NewHttpPkiAnnotator(cfg config.SdkInfo) interfaces.Annotator {
+	a := HttpPkiAnnotator{}
+	a.hash = cfg.Hash.Type
+	a.kind = contracts.AnnotationPKIHttp
+	a.sign = cfg.Signature
+	return &a
+}
+
+func (a *HttpPkiAnnotator) Do(ctx context.Context, data []byte) (contracts.Annotation, error) {
+	key := annotators.DeriveHash(a.hash, data)
+	hostname, _ := os.Hostname()
+
+	//Call parser on request
+	req := ctx.Value(testRequest)
+	parsed, err := requestParser(req.(*http.Request))
+
+	if err != nil {
+		return contracts.Annotation{}, err
+	}
+	var sig signable
+	sig.Seed = parsed
+	sig.Signature = req.(*http.Request).Header.Get("Signature")
+
+	ok, err := sig.verifySignature(a.sign.PublicKey)
```

Comment:
@husseinfakharany It appears here that we are still using the PublicKey from the config and not the key indicated by the Signature-Input header. You may want to consider extending the `signable` type defined below [like so](https://github.com/tsconn23/alvarium-sdk-go/blob/5e06fc25c15579ab84c2d0f4e5aa5aab1ccdff57/internal/annotators/http/parser.go#L26) and have the parser return an instance of `signable`. 

If you do that, then you should be able to use the `KeyId` and `Algorithm` properties to assemble the KeyInfo type that `verifySignature` expects.